### PR TITLE
chore: update loglevel in all modules

### PR DESCRIPTION
**Pull request by Shahroz16**
helps: https://linear.app/customerio/issue/MBL-105/logging-in-modules

Share logger and config among all modules and dependencies including Segment and Gist

### CODE REVIEW

**Review by Ahmed-CIO — APPROVED** [review]
LGTM... SHIP IT

File: Sources/DataPipeline/CustomerIO.swift [hunk 1]
```diff
@@ -21,6 +21,12 @@ public extension CustomerIO {
 
         let sdkConfig = SdkConfig.Factory.create(siteId: "", apiKey: "", region: .US)
         let newDiGraph = DIGraph(sdkConfig: sdkConfig)
+
+        // set the logLevel for ConsoleLogger
+        DIGraphShared.shared.logger.setLogLevel(logLevel)
+        // enable Analytics logs accordingly to logLevel
+        CustomerIO.shared.setDebugLogsEnabled(logLevel == .debug)
```
Comment (mrehan27) [inline Sources/DataPipeline/CustomerIO.swift:28]: Is there a reason for not putting this into `initialize` method? Keeping this outside `initialize` will result in `logLevel` not being set in `logger` for tests.

**Review by mrehan27 — APPROVED** [review]
Just one question, looks good otherwise 👍🏻

File: Sources/DataPipeline/CustomerIO.swift (hunk 1 above)
Comment (ami-aman) [reply Sources/DataPipeline/CustomerIO.swift:28]: [This seems to be in `initialize` method only. ](https://github.com/customerio/customerio-ios/blob/5d04ed4ef58736e54ab4b406992d4132df27044c/Sources/DataPipeline/CustomerIO.swift#L26)

File: Sources/MessagingInApp/Provider/InAppProvider.swift [hunk 2]
```diff
@@ -3,7 +3,7 @@ import Foundation
 
 // wrapper around Gist SDK to make it mockable
 protocol InAppProvider: AutoMockable {
-    func initialize(siteId: String, region: Region, delegate: GistDelegate)
+    func initialize(siteId: String, region: Region, delegate: GistDelegate, logging: Bool)
```
Comment (ami-aman) [inline Sources/MessagingInApp/Provider/InAppProvider.swift:6]: Do you plan to update the test cases for this change in a separate PR ?

**Review by ami-aman — APPROVED** [review]

File: Sources/DataPipeline/CustomerIO.swift (hunk 1 above)
Comment (Shahroz16) [reply Sources/DataPipeline/CustomerIO.swift:28]: I didn't want to rely on `DiGraph` for this as well, as i would assume we decouple away from it.

File: Sources/MessagingInApp/Provider/InAppProvider.swift (hunk 2 above)
Comment (Shahroz16) [reply Sources/MessagingInApp/Provider/InAppProvider.swift:6]: We don't have a test for initialization for `MessagingInApp`, I assume when we add that then we can check for logging too.

**Pull request merged**